auth-service/src/main/java/com/vladsimonenko/authservice/repository/impl/UserCredentialRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.vladsimonenko.authservice.repository.impl;
+
+import com.vladsimonenko.authservice.mapper.UserCredentialRowMapper;
+import com.vladsimonenko.authservice.model.UserCredential;
+import com.vladsimonenko.authservice.repository.UserCredentialRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Repository
+@RequiredArgsConstructor
+public class UserCredentialRepositoryImpl implements UserCredentialRepository {
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Override
+    public UserCredential save(UserCredential userCredential) {
+        String query = "INSERT INTO t_users (username, password)"
+                       + "VALUES (:username, :password)";
+        SqlParameterSource params = new MapSqlParameterSource()
+                .addValue("username", userCredential.getUsername())
+                .addValue("password", userCredential.getPassword());
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(query, params, keyHolder);
+
+        userCredential.setId((Integer) Objects.requireNonNull(keyHolder.getKeys()).get("id"));
+        return userCredential;
+    }
+
+    @Override
+    public Optional<UserCredential> findByUsername(String username) {
+        String query = "select * from t_users where username = :username ";
+        MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("username", username);
+        List<UserCredential> users = jdbcTemplate.query(query, params, new UserCredentialRowMapper());
+
+        return users.isEmpty() ? Optional.empty() : Optional.of(users.get(0));
```

Comment:
Обычно тернарники форматируются примерно так
```suggestion
        return users.isEmpty()
                ? Optional.empty()
                : Optional.of(users.get(0));
```